### ISSUE TITLE
One byte out of bounds access in json_parse_ex

### ISSUE THREAD

**Issue by 0n3m4ns4rmy** [json-parser/json-parser#78]
Note: Not a critical vulnerability (wouldnt even call it a vulnerability) at all but its still a bug which slows down performance and easy to fix.

When json_parse_ex checks for true, false or null it doesnt properly check the remaining length of the json string thus comparing the last character of these strings (true, false and null) with a byte beyond the given  json string.

```C
case 't':

  if ((end - state.ptr) < 3 || *(++ state.ptr) != 'r' ||
      *(++ state.ptr) != 'u' || *(++ state.ptr) != 'e')
  {
     goto e_unknown_value;
  }

  if (!new_value (&state, &top, &root, &alloc, json_boolean))
     goto e_alloc_failure;

  top->u.boolean = 1;

  flags |= flag_next;
  break;

case 'f':

  if ((end - state.ptr) < 4 || *(++ state.ptr) != 'a' ||
      *(++ state.ptr) != 'l' || *(++ state.ptr) != 's' ||
      *(++ state.ptr) != 'e')
  {
     goto e_unknown_value;
  }

  if (!new_value (&state, &top, &root, &alloc, json_boolean))
     goto e_alloc_failure;

  flags |= flag_next;
  break;

case 'n':

  if ((end - state.ptr) < 3 || *(++ state.ptr) != 'u' ||
      *(++ state.ptr) != 'l' || *(++ state.ptr) != 'l')
  {
     goto e_unknown_value;
  }

  if (!new_value (&state, &top, &root, &alloc, json_null))
     goto e_alloc_failure;

  flags |= flag_next;
  break;
```
if you would change the comparisons from < to <= this would be fixed. If you want ill send a pull request. (sorry for all the edits i got confused myself)

**Comment by harkue** [comment]
It seems be fixed with #72

**Comment by harkue** [comment]
#72
